src/call.c

```diff
@@ -2967,3 +2969,33 @@ bool call_supported(struct call *call, uint16_t tags)
 
 	return (call->supported & tags) == tags;
 }
+
+/**
+ * Get the user data for the call
+ *
+ * @param call Call object
+ *
+ * @return Call's user data
+ */
+const char *call_user_data(const struct call *call)
+{
+	return call ? call->user_data : NULL;
+}
+
+/**
+ * Set the user data of the call
+ *
+ * @param call Call object
+ * @param user_data User data to be set
+ * @return int
+ */
+
+int call_set_user_data(struct call *call, const char *user_data)
+{
+	if (!call)
+		return EINVAL;
+
+	str_dup(&call->user_data, user_data);
```

Comment:
the previous user_data should be freed with mem_deref first

`call->user_data = mem_deref(call->user_data`